android/src/main/java/com/reactnativecommunity/webview/RNCWebChromeClient.java

```diff
@@ -169,9 +174,23 @@ public void onPermissionRequest(final PermissionRequest request) {
                   androidPermission = PermissionRequest.RESOURCE_PROTECTED_MEDIA_ID;
                 }            }
             // TODO: RESOURCE_MIDI_SYSEX, RESOURCE_PROTECTED_MEDIA_ID.
+            String alertMessage = String.format("Allow this app to use your " + requestPermissionIdentifier + "?");
             if (androidPermission != null) {
                 if (ContextCompat.checkSelfPermission(this.mWebView.getThemedReactContext(), androidPermission) == PackageManager.PERMISSION_GRANTED) {
-                    grantedPermissions.add(requestedResource);
+                    AlertDialog.Builder builder = new AlertDialog.Builder(this.mWebView.getContext());
+                    builder.setMessage(alertMessage);
+                    builder.setCancelable(false);
+                    String finalAndroidPermission = androidPermission;
+                    builder.setPositiveButton("Allow", (dialog, which) -> {
+                        permissionRequest = request;
+                        grantedPermissions.add(finalAndroidPermission);
+                        requestPermissions(grantedPermissions);
+                    });
+                    builder.setNegativeButton("Don't allow", (dialog, which) -> {
+                        request.deny();
+                    });
+                    AlertDialog alertDialog = builder.create();
+                    alertDialog.show();
```

Comment:
The delayed function to accept this permissions was implemented here as well?
Would something like this worth the try?
```
this.runDelayed(() -> builder.setPositiveButton("Allow", (dialog, which) -> {
                        permissionRequest = request;
                        grantedPermissions.add(finalAndroidPermission);
                        requestPermissions(grantedPermissions);
                    }); , 500);

```